src/mlpack/methods/neighbor_search/sort_policies/nearest_neighbor_sort.hpp

```diff
@@ -150,6 +150,21 @@ class NearestNeighborSort
       return DBL_MAX;
     return a + b;
   }
+
+  /**
+   * Return the given value relaxed.
+   *
+   * @param value Value to relax.
+   * @param epsilon Relative error (non-negative).
+   *
+   * @return double Value relaxed.
+   */
+  static inline double Relax(const double value, const double epsilon)
+  {
+    if (value == DBL_MAX)
+      return DBL_MAX;
+    return (1 / (1 + epsilon)) * value;
```

Comment:
I like option 4 the most.  It is up to you completely (I have no opinion here) but another possibility would be to provide `--epsilon` for `mlpack_kfn` as well as a `-p` option that you specified, so the user could provide either.  But the disadvantage is that now there are two options instead of one. :)